### ISSUE TITLE
lexicalHTML field doesn't update in client-side Live Preview

### ISSUE THREAD

**Issue by stephtr** [payloadcms/payload#8277]
### Environment Info

```text
Binaries:
  Node: 20.9.0
  npm: 10.1.0
  Yarn: 1.22.19
  pnpm: 9.7.1
Relevant Packages:
  payload: 3.0.0-beta.107
  next: 15.0.0-canary.136
  @payloadcms/graphql: 3.0.0-beta.107
  @payloadcms/live-preview: 3.0.0-beta.107
  @payloadcms/live-preview-react: 3.0.0-beta.107
  @payloadcms/next/utilities: 3.0.0-beta.107
  @payloadcms/richtext-lexical: 3.0.0-beta.107
  @payloadcms/translations: 3.0.0-beta.29
  @payloadcms/ui/shared: 3.0.0-beta.107
  react: 19.0.0-rc-e56f4ae3-20240830
  react-dom: 19.0.0-rc-e56f4ae3-20240830
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.0.0: Mon Aug 12 20:52:41 PDT 2024; root:xnu-11215.1.10~2/RELEASE_ARM64_T6031
  Available memory (MB): 36864
  Available CPU cores: 14
```

### Describe the Bug

When adding a **rich text field** and a **lexicalHTML field** for automatically converting it into HTML code, as well as enabling **client-side Live Preview**, only the rich text field content gets updated while typing, but not the generated HTML field.

Is this a bug or a design decision?
If a bug, I could quickly create a repo for reproduction.

### Reproduction Steps

* Create a collection with a rich text field
* Add a lexicalHTML field [as documented](https://payloadcms.com/docs/beta/lexical/converters#outputting-html-from-the-collection)
* Add client-side Live Preview [as documented](https://payloadcms.com/docs/beta/live-preview/client)

**Comment by alexnitta** [comment]
I'm also facing this issue and curious how it might be resolved; the experience for editing a lexicalHTML field is less than ideal in the current state of the live preview feature.

**Comment by stephtr** [comment]
I guess you anyway know, but you could also switch to [server-side live-preview](https://payloadcms.com/docs/beta/live-preview/server) together with [Versions/Draft-mode](https://payloadcms.com/docs/beta/versions/drafts) and [Autosave](https://payloadcms.com/docs/beta/versions/autosave). Overall, I find this the even better experience.

**Comment by alexnitta** [comment]
@stephtr I was not aware of that option; I'll look into it. Thanks for the tip!

**Comment by alexnitta** [comment]
@stephtr to be clear, are you saying that we should see the generated HTML field update after an autosave? I tried setting up server-side live preview with versions/draft-mode and autosave, but I didn't see the generated HTML update in the live preview. I'm on an outdated version of the Payload beta, so that could be my issue, but I wasn't actually sure if you expected this to fix the problem.

**Comment by afreidz** [comment]
facing the same issue on payload version: 2.30.1. I am guessing the afterRead hook that lexicalHTML uses behind the scenes is not re-running before the updated data is passed back to the live server client.

**Comment by stephtr** [comment]
@alexnitta yes, for me that worked (I was running beta 107). You then either have about 2 seconds delay until autosave kicks in or decrease the autosave interval

**Comment by namitasai** [comment]
I'm facing same issue, the lexicalHTML is not callable on [slug]/page.ts .... steps: 

  fields: [
    {
      name: "title",
      type: "text",
      required: true,
      label: "Post Title",
      hooks: {
        beforeValidate: [
          ({ value, data, operation }: FieldHookArgs<PostType>) => {
            if (operation === "create" && data && (!data.slug || data.slug === "") && value) {
              data.slug = value
                .toLowerCase()
                .replace(/\s+/g, "-")
                .replace(/[^a-z0-9-]/g, "");
            }
            return value;
          },
        ],
      },
    },
    {
      name: "slug",
      type: "text",
      required: false,
      unique: false,
      admin: {
        description: "Auto-generated URL path",
      },
    },
    {
      name: "lexicalContent",
      type: "richText",
      editor: lexicalEditor({
        features: ({ defaultFeatures }) => [
          ...defaultFeatures,
          HTMLConverterFeature({}),

          LinkFeature({
            fields: ({ defaultFields }) => [
              ...defaultFields,
              {
                name: "rel",
                label: "Rel Attribute",
                type: "select",
                hasMany: true,
                options: ["noopener", "noreferrer", "nofollow"],
                admin: {
                  description: "Define relationship attributes for links.",
                },
              },
            ],
          }),

          UploadFeature({
            collections: {
              uploads: {
                fields: [
                  {
                    name: "caption",
                    type: "richText",
                    editor: lexicalEditor(),
                  },
                ],
              },
            },
          }),
          BlocksFeature({
            blocks: [Banner, CallToAction, ImageGrid],
          }),
        ],
      }),

      required: false,
      label: "Post Content",

      admin: {
        description: "Add the main content of the post.",
      },
    },

 lexicalHTML("lexicalContent", { name: "lexicalContentConverted_html" }),

--> But when I try to call this on page.ts, it doesn't even recognize this, Don't even know how to call/pass this lexicalHTML, lexicalContentConverted_html to make it work, as this is not even recognized as props or anything like that. Please give code on how to use lexicalHTML, for live-preview in lexical format for the post.content field. Thank you. Waiting for reply eagerly.

**Comment by namitasai** [comment]
#> I am also facing this issue, however I don't have the option to switch to server-side live preview, because I am using Remix on the frontend. And I cannot use JSON, because the RichText response can contain lots and lots of deeply nested relation and link objects ... Would be cool, if client-side preview would enable this option, or maybe we can switch to JSON when doing preview and use HTML when doing preview

@HriBB try to do serialization of the fields where you are using lexicalEditor, that generates Lexical JSON, which will not accept Rich HTML format or strings, so do serialization for those payload fields such as content, title, excerpt, blocks, upload or whatever else you are using, then you can use those serialiazed fields on client side also or preview. Basically PayloadCMS should fix their HTMLConverter(), LexicalHTML(), so we are making workarounds by writing so much code, it is a lot of work.  Steps for you: Define those fields in Collection Posts for example, as richText, LexicalEditor() (2) use hooks: beforeChange() -- write your logic to trigger conversion of lexicalJSON to HTML before saving to db 3) call it in [slug]/page.tsx   NOTE: Caveat: it is not as easy, because you also need to write conversion logic like LexicalRender functions or RichTextRender whatever you can call it, then pass it in hooks, but i hope I've given you some idea, _PLEASE DON'T MIND @PAYLOADCMS: I TRULY WISH, PAYLOAD CMS CAN IMPROVE THEIR DOCUMENTATION, GIVE WORKING EXAMPLES FOR EACH FEATURE THEY LAUNCH BY GIVING END TO END EXAMPLES ON GITHUB THAN LEAVING US FOR DOING HIT AND TRY PLAYING WITH EACH AND EVERY COMMAND OF PAYLOAD CMS. I TRIED TO CONTACT THEM VIA THEIR FORM AS NO PHONE NUMBER IS AVAILABLE; SOMEONE CONTACTED ME ON THIS TO GET MORE INFORMATION BUT THEY NEVER KEPT ME INFORMED ON THE STATUS FOR THIS OR PROVIDED WORKAROUND OR CONNECTED ME WITH THE TECH ON THIS.  YOU CAN TRY TO CONTACT THEM ON THIS AS WELL._

**Comment by GermanJablo** [comment]
The issue https://github.com/payloadcms/payload/issues/8168 describes the same problem from a more technical perspective.

I just wrote a comment there explaining how this is actually mostly a documentation issue and the recommended ways to resolve it. I'll soon improve the documentation with the information described in that comment.

Thanks everyone for your patience, and if you have any questions, let me know!

**Comment by AlessioGr** [comment]
Fixed by https://github.com/payloadcms/payload/pull/13619 - the lexicalHTML field is now supported in client-side live preview!
